### ISSUE TITLE
webconnectivity: handle down websites such as doh.centraleu.pi-dns.com

### ISSUE THREAD

**Issue by bassosimone** [ooni/probe#2299]
Let us start our journey with the obligatory [MAT search](https://explorer.ooni.org/chart/mat?test_name=web_connectivity&domain=doh.centraleu.pi-dns.com&since=2021-09-12&until=2022-09-12&axis_x=measurement_start_day):

<img width="1256" alt="Screenshot 2022-09-13 at 08 26 17" src="https://user-images.githubusercontent.com/337298/189826025-e3c60bf3-9d4e-4ff7-a608-0e105448f7cb.png">

So, judging from the MAT search, this website presents anomalies. But, is that really the case? Could it be that there is a bug in Web Connectivity v0.4.x that causes the website to erroneously being flagged as anomaly?

Judging from [this v0.4.1 measurement](https://explorer.ooni.org/measurement/20220911T214558Z_webconnectivity_US_13335_n1_fk64jl5M3zIGGTM7?input=https%3A%2F%2Fdoh.centraleu.pi-dns.com%2Fdns-query%3Fdns%3Dq80BAAABAAAAAAAAA3d3dwdleGFtcGxlA2NvbQAAAQAB) maybe we're misflagging the results here. The connections indeed all fail for both the probe and the TH, so this should be flagged as "website down", at least as far as that measurement I showed was concerned. But [also this measurement shows similar issues](https://explorer.ooni.org/measurement/20220911T221129Z_webconnectivity_TW_9916_n1_tDd7wjWODdx6eMR1?input=https%3A%2F%2Fdoh.centraleu.pi-dns.com%2Fdns-query%3Fdns%3Dq80BAAABAAAAAAAAA3d3dwdleGFtcGxlA2NvbQAAAQAB). [And also this one](https://explorer.ooni.org/measurement/20220911T235008Z_webconnectivity_AU_4764_n1_Ea9Z9PDsU74rsg8U?input=https%3A%2F%2Fdoh.centraleu.pi-dns.com%2Fdns-query%3Fdns%3Dq80BAAABAAAAAAAAA3d3dwdleGFtcGxlA2NvbQAAAQAB). So maybe I have a point here 🥶.

The approach taken by v0.5 LTE for this kind of situations is a bit more correct. [In this measurement](https://explorer.ooni.org/measurement/20220911T105037Z_webconnectivity_IT_30722_n1_ruzuQ219SmIO9SrT?input=https://doh.centraleu.pi-dns.com/dns-query?dns=q80BAAABAAAAAAAAA3d3dwdleGFtcGxlA2NvbQAAAQAB), it says "look, mate, I don't know", and sets .Blocking to null and .Failure to null, thus making it an errored measurement. However, the best approach in this case would be to recognize that the website is just down and flag is as such like we did for DNS a few commits ago in https://github.com/ooni/probe-cli/commit/b10eea47e7ea1d66b7e06c1894591d540900fbc8.

**Comment by bassosimone** [comment]
After https://github.com/ooni/probe-cli/pull/957, [webconnectivity LTE flags this website as down as it ought to be](https://explorer.ooni.org/measurement/20220913T064929Z_webconnectivity_IT_30722_n1_MP4DAOIzAJGxHpFE?input=https://doh.centraleu.pi-dns.com/dns-query?dns=q80BAAABAAAAAAAAA3d3dwdleGFtcGxlA2NvbQAAAQAB)

**Comment by bassosimone** [comment]
So, we're fixing this for v0.5 inside https://github.com/ooni/probe-cli/pull/957. It remains to be seen whether we also want to apply a similar fix to the v0.4 implementation, or whether we prefer focusing on v0.5. I'll chat with @hellais to get a second opinion about this topic. I'd ideally like to get more experience with v0.5 and then try to bring some changes back to v0.4 in a batch, because it might be more efficient to do that.

**Comment by bassosimone** [comment]
I just added the [fixed-by-webconnectivity-lte](https://github.com/ooni/probe/labels/fixed-by-webconnectivity-lte) flag such that we can flag all the issues (such as this one), that are open but will automatically be fixed when we switch over to the new implementation of Web Connectivity. I am now going to tag similar issues. Because the development of the new version of Web Connectivity took a long time, we need a way to distinguish between issues that have been fixed in the next version of Web Connectivity and issues that have not been fixed yet.
